toolboxes/core/gpu/cuNDArray_blas.h

```diff
@@ -20,31 +20,31 @@
 
 namespace Gadgetron{
 
-  template<class T> T dot( cuNDArray<T> *x, cuNDArray<T> *y, bool cc = true );
+  template<class T> T dot( cuNDArray<T> *x, cuNDArray<T> *y, size_t batchSize = INT_MAX, bool cc = true );
 
-  template<class T> typename realType<T>::Type nrm2( cuNDArray<T> *x );
+  template<class T> typename realType<T>::Type nrm2( cuNDArray<T> *x , size_t batchSize = INT_MAX);
 
-  template<class T> void axpy( T a, cuNDArray<T>* x, cuNDArray<T>* y );
+  template<class T> void axpy( T a, cuNDArray<T>* x, cuNDArray<T>* y , size_t batchSize = INT_MAX);
 
-  template<class T> void axpy( T a, cuNDArray<complext<T> > *x, cuNDArray<complext<T> > *y );
+  template<class T> void axpy( T a, cuNDArray<complext<T> > *x, cuNDArray<complext<T> > *y , size_t batchSize = INT_MAX);
   
   /**
    * @brief Gets the index of the index of the element with minimum absolute
    * @param x Input data
    * @return index of absolute minimum values
    * @details Note that this returns the C-style index and NOT the Fortran index.
    */
-  template<class T> size_t amin( cuNDArray<T> *x );
-  
+  template<class T> size_t amin( cuNDArray<T> *x , size_t batchSize = INT_MAX);
```

Comment:
Please update the function documentations to include the `batchSize`

Reply:
Done - I added function documentation for other functions as well.